lib/createJestRunner.ts

```diff
@@ -1,20 +1,44 @@
-import throat from 'throat';
+import type * as JestResult from '@jest/test-result';
+import type { Config } from '@jest/types';
+import type * as JestRunner from 'jest-runner';
 import Worker from 'jest-worker';
+import throat from 'throat';
+import type { Path, TestRunner } from './types';
 
 class CancelRun extends Error {
-  constructor(message) {
+  constructor(message?: string) {
     super(message);
     this.name = 'CancelRun';
   }
 }
 
-const createRunner = (runPath, { getExtraOptions } = {}) => {
-  class BaseTestRunner {
-    constructor(globalConfig) {
+export default function createRunner<
+  ExtraOptionsType extends Record<string, unknown>
+>(
+  runPath: Path,
+  { getExtraOptions }: { getExtraOptions?: () => ExtraOptionsType } = {},
+): typeof TestRunner {
+  return class BaseTestRunner implements TestRunner {
+    readonly _globalConfig: Config.GlobalConfig;
+
+    readonly _context: JestRunner.TestRunnerContext;
```

Comment:
```suggestion
    private readonly _globalConfig: Config.GlobalConfig;

    private readonly _context: JestRunner.TestRunnerContext;
```

Reply:
Making `_globalConfig` and `_context` private requires removing the corresponding class properties in the `declare class TestRunner { /* */ }` otherwise typescript complains about `Types have separate declarations of a private property '_globalConfig'.`

I can add the `private` keyword if you think [this](https://github.com/jest-community/create-jest-runner/pull/40#discussion_r517761111) is good enough :)